### PR TITLE
Issue #22: confidence thresholds config and enforcement

### PR DESCRIPTION
**Pull request by stranske**
> **Source:** Issue #22

#### Scope
Confidence policy determines what can flow straight through and what must be reviewed; this must be explicit and configurable from day one.

### Related Issues/PRs
- [#9](https://github.com/stranske/Inv-Man-Intake/issues/9)
- [#7](https://github.com/stranske/Inv-Man-Intake/issues/7)

#### Tasks
- [ ] Implement confidence scoring utility and schema attachment
- [ ] Implement config loader for threshold values
- [ ] Enforce approved v1 thresholds in orchestration decisions
- [ ] Add tests for threshold boundary behavior
- [ ] Document threshold semantics for operator tuning

#### Acceptance criteria
- [ ] Field-level and document-level thresholds are read from config
- [ ] Boundary values are handled deterministically in tests
- [ ] Escalation trigger activates for mandatory fields below threshold
- [ ] Documentation describes each threshold effect clearly

**Head SHA:** 8c172eff007a9b1f276be7a0874459c07a1e1e61
**Latest Runs:** ⏭️ skipped — Agents PR Meta
**Required:** gate: ⏸️ not started

| Workflow / Job | Result | Logs |
|----------------|--------|------|
| Agents PR Event Hub | ❔ in progress | [View run](https://github.com/stranske/Inv-Man-Intake/actions/runs/22835732663) |
| Agents PR Meta | ⏭️ skipped | [View run](https://github.com/stranske/Inv-Man-Intake/actions/runs/22835732633) |

### CODE REVIEW

**Comment by stranske** [comment]
Rebased onto latest main and force-pushed.
Local validation after rebase: 263 tests passed.

**Comment by stranske** [comment]
[issue-completion-audit][ica-nonpass-followup] Disposition for verify:compare non-PASS on PR #86:

- Determination: not-warranted for code fix (documentation-only outcome).
- Summary concern: non-PASS indicated missing disposition documentation, not a behavioral defect.
- Rationale: documenting disposition closes the verification traceability gap without additional product changes.
- Tracking artifact: issue #112; disposition notes recorded in `docs/reverification/pr-86-verify-compare-output.md` on branch `codex/issue-112`.

No additional fix PR is required for this specific non-PASS output.

File: docs/contracts/extraction_thresholds.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Extraction Thresholds (v1)
+
+Issue: #22
+
+## Config Source
+
+Thresholds are loaded from [`config/extraction_thresholds.yaml`](../../config/extraction_thresholds.yaml) using `load_threshold_config` in `src/inv_man_intake/extraction/confidence.py`.
+
```
[the remark this answers is not in the document]
Comment (stranske) [reply docs/contracts/extraction_thresholds.md:13]: Dispositioned as not-warranted for additional follow-up. Current `main` now documents the restricted config subset explicitly, including the required block-list form for `mandatory_fields`, rejection of inline list syntax, and rejection of unknown keys. That resolves the operator-facing ambiguity this thread identified, so no separate remediation PR is needed beyond audit closure.

File: src/inv_man_intake/extraction/confidence.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            in_mandatory_list = True
+            continue
+        if in_mandatory_list and line.startswith("-"):
+            mandatory_fields.append(line.removeprefix("-").strip().strip('"'))
+            continue
+        if ":" in line:
+            in_mandatory_list = False
+            key, value = line.split(":", 1)
+            values[key.strip()] = value.strip().strip('"')
+
```
[the remark this answers is not in the document]
Comment (stranske) [reply src/inv_man_intake/extraction/confidence.py:67]: Dispositioned as not-warranted for additional follow-up. Current `main` already takes the safer path requested here: `load_threshold_config()` raises on inline `mandatory_fields` forms and on unknown keys instead of silently accepting them. Replacing the parser wholesale would be discretionary refactoring, not required follow-up to close this review concern.

File: src/inv_man_intake/extraction/confidence.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    )
+
+    eligible_key_fields = [
+        key
+        for key in key_fields
+        if key in field_by_key and field_by_key[key].confidence >= config.key_field_confidence_min
+    ]
+    key_field_coverage_ratio = (
+        len(eligible_key_fields) / len(key_fields) if key_fields else 1.0
+    )
```
[the remark this answers is not in the document]
Comment (stranske) [reply src/inv_man_intake/extraction/confidence.py]: Dispositioned as not-warranted for additional follow-up. The risky behavior is already gone on `main`: empty `key_fields` now produces `0.0` coverage and forces escalation instead of auto-passing the document. This thread is effectively outdated, so the remaining work is audit closure, not another fix PR.

File: tests/extraction/test_thresholds.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                source_page=1,
+            )
+            for key, confidence in fields
+        ),
+    )
+
+
+def test_load_threshold_config_reads_policy_values() -> None:
+    config = load_threshold_config(Path("config/extraction_thresholds.yaml"))
+
```
[the remark this answers is not in the document]
Comment (stranske) [reply tests/extraction/test_thresholds.py:36]: Dispositioned as not-warranted for additional follow-up. Current `main` already builds the config path from `Path(__file__).resolve().parents[2] / "config" / "extraction_thresholds.yaml"`, so the tests are no longer CWD-dependent. The concern was valid when raised, but it does not require any new implementation work now.
